control/webapp/admin.py

```diff
@@ -78,16 +79,24 @@ def status(id):
     notes = [x for x in log if x.type == "note"]
 
     job_home_url = url_for('admin.view_jobs', state=job.state)
-    for_society = isinstance(job, SocietyJob) and job.owner.crsid != utils.auth.principal
-    owner_in_context = job.society_society if isinstance(job, SocietyJob) else job.owner_crsid
+    for_society = isinstance(job, SocietyJob)
+
+    owner_in_context = job.owner_crsid
+    danger = []
+    if job.owner.danger:
+        danger.append(job.owner_crsid)
+    if for_society:
+        owner_in_context = job.society_society
+        if job.society and job.society.danger:
+            danger.append(job.society_society)
 
     return render_template("admin/status.html", job=job, notes=notes, log=log, job_home_url=job_home_url,
-                           for_society=for_society, owner_in_context=owner_in_context,
+                           for_society=for_society, owner_in_context=owner_in_context, danger=danger,
                            principal=utils.raven.principal, has_create_log=has_create_log)
 
 
 _actions = {
```

Comment:
Could this variable do with a better name, or a comment to explain what the dict values are?  I can't tell by looking, although they seem like they're "state pathways" the control panel may push jobs along.

_Side note that this logic belongs in job code, not frontend code_

Reply:
Looking back at this, I think I may have over-optimised the pass-through of those flags.  Though a fair side-note, it probably wants to be attributes of the `Job` class.

Reply:
These have now been moved to `Job`.